allow_price_update_on_locked_so/views/sale_order_views.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sale.order.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//page[@name='order_lines']/field[@name='order_line']/tree/field[@name='price_unit']"
+                position="after"
+            >
+                <field name="state" invisible="1" />
+                <button
+                    title="Edit Price Unit"
+                    class="btn btn-secondary fa fa-pencil"
+                    attrs="{'invisible':[('state', 'not in', ['done', 'cancel'])]}"
+                    groups="allow_price_update_on_locked_so.group_edit_unit_price_on_sol"
+                />
+            </xpath>
+            <xpath
+                expr="//page[@name='order_lines']/field[@name='order_line']"
+                position="attributes"
+            >
+                <attribute name="attrs">
+                    {"readonly": [("state", "=", "cancel")]}
+                </attribute>
+            </xpath>
+            <xpath
+                expr="//page[@name='order_lines']/field[@name='order_line']/tree/field[@name='name']"
+                position="attributes"
+            >
+                <attribute name="attrs">
+                    {"readonly": [("state", "in", ["done", "cancel"])]}
+                </attribute>
+            </xpath>
+            <xpath
+                expr="//page[@name='order_lines']/field[@name='order_line']/tree/field[@name='product_uom_qty']"
+                position="attributes"
+            >
+                <attribute name="attrs">
+                    {"readonly": [("state", "in", ["done", "cancel"])]}
+                </attribute>
+            </xpath>
+            <xpath
+                expr="//page[@name='order_lines']/field[@name='order_line']/tree/field[@name='qty_delivered']"
+                position="attributes"
+            >
+                <attribute name="attrs">
+                    {"readonly": [("state", "in", ["done", "cancel"])]}
+                </attribute>
+            </xpath>
+            <xpath
+                expr="//page[@name='order_lines']/field[@name='order_line']/tree/field[@name='tax_id']"
+                position="attributes"
+            >
+                <attribute name="attrs">
+                    {
+                        'readonly': [
+                            '|',
+                            ('state', 'in', ('done', 'cancel')),
+                            ('qty_invoiced', '&gt;', 0)
+                        ]
+                    }
+                </attribute>
+            </xpath>
+        </field>
+    </record>
+
+</odoo>
```

Comment:
As discussed:
- I suggest the new way to implement the feature: click the button to open the new wizard --> set new price --> update so line

With your implemented:
- New potential issues: The other is installed, but your implementations on this one haven't been taken into account. the blocking on view won't work